piwik-sdk/src/test/java/org/piwik/sdk/TrackerTest.java

```diff
@@ -808,6 +809,19 @@ public void testPreviousVisit() throws Exception {
         assertEquals(1000l, Long.parseLong(queryParams.get(QueryParams.PREVIOUS_VISIT_TIMESTAMP)));
     }
 
+    @Test
+    public void testSetCustomDimensions() throws Exception {
+        CustomDimensions customDimensions = new CustomDimensions();
+        customDimensions.set(1, "foo");
+        customDimensions.set(2, "bar");
+        customDimensions.set(3, "empty").set(3, null);
```

Comment:
An empty string behaves the same way as `null` value
https://github.com/piwik/plugin-CustomDimensions/blob/e330cb2c0db08d1a5b4e489d3fba7acc0e7cc43c/Tracker/CustomDimensionsRequestProcessor.php#L100